src/core/csv/reader_fread.cc

```diff
@@ -296,6 +297,7 @@ void FreadReader::detect_sep_and_qr() {
         if (thisncol < 0) {
           // invalid file with this sep and quote rule; abort
           numFields[0] = -1;
+          if (quoteRule <= 1) max_quoteRule = 2+(quoteRule*2);
```

Comment:
so if parse fails (`thisncol<0`) at `quoteRule=0` (but not at `quoteRule=1`), then the new  `max_quoteRule` will be 2, which is the same as the initial  value of `max_quoteRule`. I'm not sure i fully understand the logic  here.

Reply:
My apologies for leaving this issue open for a long time. This logic is obviously sloppy and I have pushed another commit, kindly have a look. 

Also, the tests seems to fail on py36 and 37 but it passes on my Linux machine when I tested it on py36, py37 and py38. I have no permission to check which/why tests are failing, can you have a look? Thanks.

Reply:
@st-pasha can you kindly look into why this is failing?

Reply:
Ah yes, it fails for the same reason that #2708 was failing. So if you merge `main` into this branch, the errors should (hopefully) go away.

Reply:
Done